lib/util/string_conv/string_conv.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ctype.h>
+#include <string.h>
+#include <stdlib.h>
+#include <logging/log.h>
+
+#define LOG_MODULE_NAME string_conv
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_STRING_CONV);
+
+int string_conv_str2long(char *str, int *val)
+{
+	int str_len = strlen(str);
+	int temp_val;
+	int idx = 0;
+
+	if (!strncmp(str, "-", 1)) {
+		idx = 1;
+	}
+
+	for (int i = idx; i < str_len; i++) {
+		if (isdigit((int)str[i])) {
+			continue;
+		}
+
+		LOG_ERR("\"%s\" is not recognized as a valid signed decimal input", str);
```

Comment:
I'm not sure we should log an error here. I think it would be better is this was a pure function.

Reply:
You mean log at all, or in this particular function?